src/drive/web/modules/actions/utils.js

```diff
@@ -0,0 +1,159 @@
+        .getDownloadLinkById(file.id)
+      const filename = file.name
+
+      forceFileDownload(
+        `${client.getStackClient().uri}${downloadURL}?Dl=1`,
+        filename
+      )
+    } catch (error) {
+      Alerter.error(downloadFileError(error))
+      throw error
+    }
+  } else {
+    const ids = files.map(f => f.id)
+    const href = await client
+      .collection('io.cozy.files')
+      .getArchiveLinkByIds(ids)
+    const fullpath = `${client.getStackClient().uri}${href}`
+    forceFileDownload(fullpath, 'files.zip')
+  }
+}
+
+const isAlreadyInTrash = err => {
+  const reasons = err.reason !== undefined ? err.reason.errors : undefined
+  if (reasons) {
+    for (const reason of reasons) {
+      if (reason.detail === 'File or directory is already in the trash') {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+export const trashFiles = async (files, client) => {
+  try {
+    for (const file of files) {
+      await client.collection('io.cozy.files').destroy(file)
+      client.collection('io.cozy.permissions').revokeSharingLink(file)
+    }
+
+    Alerter.success('alert.trash_file_success')
+  } catch (err) {
+    if (!isAlreadyInTrash(err)) {
+      Alerter.error('alert.try_again')
+    }
+  }
+}
+
+export const exportFilesNative = async (files, client, filename) => {
+  const downloadAllFiles = files.map(async file => {
+    const response = await client
+      .collection('io.cozy.files')
+      .fetchFileContent(file)
+
+    const blob = await response.blob()
+    const filenameToUse = filename ? filename : file.name
+    const localFile = await saveFileWithCordova(blob, filenameToUse)
+    return localFile.nativeURL
+  })
+
+  try {
+    Alerter.info('alert.preparing', {
+      duration: Math.min(downloadAllFiles.length * 2000, 6000)
+    })
+    const urls = await Promise.all(downloadAllFiles)
+    if (urls.length === 1 && isIOS()) {
+      //TODO
+      //It seems that files: is not well supported on iOS. url seems to work well
+      //at with one file. Need to check when severals
+      window.plugins.socialsharing.shareWithOptions(
+        {
+          url: urls[0]
+        },
+        result => {
+          if (result.completed === true) {
+            Alerter.success('mobile.download.success')
+          }
+        },
+        error => {
+          throw error
+        }
+      )
+    } else {
+      window.plugins.socialsharing.shareWithOptions(
+        {
+          files: urls
+        },
+        null,
+        error => {
+          throw error
+        }
+      )
+    }
+  } catch (error) {
+    Alerter.error(downloadFileError(error))
+  }
+}
+
+export const openFileWith = async (file, client, filename) => {
+  if (isMobileApp() && window.cordova.plugins.fileOpener2) {
+    let fileData
+    try {
+      fileData = await client.collection('io.cozy.files').fetchFileContent(file)
+    } catch (error) {
+      Alerter.error(openFileDownloadError(error))
+      throw error
```

Comment:
Why do we rethrow the error here ?

Reply:
Good catch, I don't think it has any business being here anymore.